### PR TITLE
refactor(contracts): do not return contract addresses on poll deploy

### LINKED ISSUE #1724: Maci.deployPoll no need to return PollAddr as they are stored and do not emit on event
- [x] do not return pollAddr as we are storing them so it's redundant 
- [x] do not emit pollAddr in event, only pollId, mode and coordinator key - subgraph can take  the pollId and pull the contract addresses from storage vs from event

### PR DESCRIPTION
**Pull request by ctrlc03**
# Description

Do not return poll/mp/tally addresses as they are already saved to state

## Related issue(s)

fix #1724 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

**Review by kittybest — APPROVED** [review]
Thanks!

File: contracts/ts/genMaciState.ts [hunk 1]
```diff
@@ -82,24 +82,27 @@ export const genMaciStateFromContract = async (
       });
     });
 
-    deployPollLogs.forEach((event) => {
+    // eslint-disable-next-line @typescript-eslint/prefer-for-of
+    for (let j = 0; j < deployPollLogs.length; j += 1) {
```
Comment (0xmad) [inline contracts/ts/genMaciState.ts:86]: Why did you change it here?

File: subgraph/tests/common.ts [hunk 2]
```diff
@@ -22,3 +22,18 @@ export function mockPollContract(): void {
...
+    Address.fromString("0xA16081F360e3847006dB660bae1c6d1b2e17eC2A"),
+    "getPoll",
+    "getPoll(uint256):((address,address,address))",
+  )
+    .withArgs([ethereum.Value.fromUnsignedBigInt(BigInt.fromI32(1))])
+    .returns([
+      ethereum.Value.fromAddress(DEFAULT_POLL_ADDRESS),
+      ethereum.Value.fromAddress(DEFAULT_MESSAGE_PROCESSOR_ADDRESS),
+      ethereum.Value.fromAddress(DEFAULT_TALLY_ADDRESS),
+    ]);
```
Comment (0xmad) [inline subgraph/tests/common.ts:38]: I think it can be a reason of the error

```suggestion
    .returns(
      ethereum.Value.fromAddress(DEFAULT_POLL_ADDRESS),
      ethereum.Value.fromAddress(DEFAULT_MESSAGE_PROCESSOR_ADDRESS),
      ethereum.Value.fromAddress(DEFAULT_TALLY_ADDRESS),
    );
```

**Review by 0xmad — COMMENTED** [review]
@ctrlc03 thanks, just some comments

File: subgraph/tests/common.ts (hunk 2 above)
Comment (0xmad) [reply subgraph/tests/common.ts:38]: No, it's correct way to mock tuples

File: contracts/ts/genMaciState.ts (hunk 1 above)
Comment (ctrlc03) [reply contracts/ts/genMaciState.ts:86]: because we are calling a async function to get the contract addresses, and async doesn't play well with forEach
